### ISSUE TITLE
Please add model maker support macos

### ISSUE THREAD

**Issue by lucasjinreal** [google/mediapipe#4088]
### OS Platform and Distribution

macos

### Programming Language and version

python

### Android AAR (if android)

None

### Describe the problem

```shell
A bug happened!


pip install mediapipe-model-maker==0.1.0.2                  main
Looking in indexes: https://pypi.tuna.tsinghua.edu.cn/simple, https://pypi.ngc.nvidia.com
Collecting mediapipe-model-maker==0.1.0.2
  Downloading https://pypi.tuna.tsinghua.edu.cn/packages/15/51/a116ce4c42224eaf9f8d193f53e9b951a4b8e5cd90a3e403145e299cd59e/mediapipe_model_maker-0.1.0.2-py3-none-any.whl (7.2 MB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 7.2/7.2 MB 2.1 MB/s eta 0:00:00
ERROR: Could not find a version that satisfies the requirement mediapipe==0.9.0.1 (from mediapipe-model-maker) (from versions: none)
ERROR: No matching distribution found for mediapipe==0.9.0.1
```

### Complete Logs

```shell
pip install mediapipe-model-maker==0.1.0.2                  main
Looking in indexes: https://pypi.tuna.tsinghua.edu.cn/simple, https://pypi.ngc.nvidia.com
Collecting mediapipe-model-maker==0.1.0.2
  Downloading https://pypi.tuna.tsinghua.edu.cn/packages/15/51/a116ce4c42224eaf9f8d193f53e9b951a4b8e5cd90a3e403145e299cd59e/mediapipe_model_maker-0.1.0.2-py3-none-any.whl (7.2 MB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 7.2/7.2 MB 2.1 MB/s eta 0:00:00
ERROR: Could not find a version that satisfies the requirement mediapipe==0.9.0.1 (from mediapipe-model-maker) (from versions: none)
ERROR: No matching distribution found for mediapipe==0.9.0.1
```

**Comment by ayushgdev** [comment]
@jinfagang 
Thanks for digging out this issue. Would you please confirm if the Mac you are using is ARM or intel based?

**Comment by jinfagang** [comment]
arm, I specifcally point out mediapipe-sillicon, this version lagging for a long range of new versions.

**Comment by ayushgdev** [comment]
Hello @jinfagang mediapipe-silicon is not an official package by Google. You can check out [this comment](https://github.com/google/mediapipe/pull/2591#issuecomment-1105351727). 
We are yet to release the official support of MediaPipe for apple silicon PCs.

**Comment by jinfagang** [comment]
@ayushgdev does there a time scheduler when will M1 mac support officially?

**Comment by ayushgdev** [comment]
We don't have any timeline yet @jinfagang

**Comment by cikichen** [comment]
```
INFO: pip is still looking at multiple versions of mediapipe-model-maker to determine which version is compatible with other requirements. This could take a while.
ERROR: Cannot install mediapipe-model-maker==0.1.0.2, mediapipe-model-maker==0.1.1.0 and mediapipe-model-maker==0.1.1.1 because these package versions have conflicting dependencies.

The conflict is caused by:
    mediapipe-model-maker 0.1.1.1 depends on mediapipe==0.9.2.1
    mediapipe-model-maker 0.1.1.0 depends on mediapipe==0.9.2.1
    mediapipe-model-maker 0.1.0.2 depends on mediapipe==0.9.0.1

To fix this you could try to:
1. loosen the range of package versions you've specified
2. remove package versions to allow pip attempt to solve the dependency conflict

ERROR: ResolutionImpossible: for help visit https://pip.pypa.io/en/latest/topics/dependency-resolution/#dealing-with-dependency-conflicts
```
on mac m1

**Comment by BjarneHerland** [comment]
Same here on Mac M1.

Note that the "older" [tflite-model-maker](https://pypi.org/project/tflite-model-maker/) does not install on Arm-based Macs either and that we are [referred here](https://github.com/tensorflow/tensorflow/issues/60431#issuecomment-1671877743) as the "workaround".

Please clean up dependencies to let us use these otherwise nice frameworks! I'd hate to be forced to PyTorch just because of this! 😞

**Comment by joezoug** [comment]
@cikichen ,

Can you try to install the latest version of [mediapipe-model-maker==0.2.1.3](https://pypi.org/project/mediapipe-model-maker/0.2.1.3/)? Mediapipe now supports M1 mac so the error message you are seeing should not apply anymore for the latest release.

**Comment by twiceyuan** [comment]
@joezoug 
The latest version 0.2.1.3 on my device (Apple Silicon M1 Pro) still cannot be installed. The error message is:

```
Collecting tf-models-official>=2.13.1 (from mediapipe-model-maker==0.2.1.3)
  Using cached tf_models_official-2.14.2-py2.py3-none-any.whl.metadata (1.4 kB)
  Using cached tf_models_official-2.14.1-py2.py3-none-any.whl.metadata (1.4 kB)
  Using cached tf_models_official-2.14.0-py2.py3-none-any.whl.metadata (1.4 kB)
  Using cached tf_models_official-2.13.2-py2.py3-none-any.whl.metadata (1.4 kB)
  Using cached tf_models_official-2.13.1-py2.py3-none-any.whl.metadata (1.4 kB)
Collecting pyyaml<5.4.0,>=5.1 (from tf-models-official>=2.13.1->mediapipe-model-maker==0.2.1.3)
  Using cached PyYAML-5.3.1.tar.gz (269 kB)
  Preparing metadata (setup.py) ... done
ERROR: Cannot install mediapipe-model-maker because these package versions have conflicting dependencies.

The conflict is caused by:
    tf-models-official 2.15.0 depends on tensorflow-text~=2.15.0
    tf-models-official 2.14.2 depends on tensorflow-text~=2.14.0
    tf-models-official 2.14.1 depends on tensorflow-text~=2.14.0
    tf-models-official 2.14.0 depends on tensorflow-text~=2.14.0
    tf-models-official 2.13.2 depends on tensorflow-text~=2.13.0
    tf-models-official 2.13.1 depends on tensorflow-text~=2.13.0

To fix this you could try to:
1. loosen the range of package versions you've specified
2. remove package versions to allow pip attempt to solve the dependency conflict

ERROR: ResolutionImpossible: for help visit https://pip.pypa.io/en/latest/topics/dependency-resolution/#dealing-with-dependency-conflicts
```

My python version is 3.10.13

**Comment by BjarneHerland** [comment]
@cikichen thanks for the hint! However, the problem seems to boil down to the dependency `tensorflow-text==2.15.0` which does not provide aarch64-wheels for neither manylinux nor macos, see

https://pypi.org/project/tensorflow-text/2.15.0/#files

**Comment by joezoug** [comment]
@BjarneHerland thank you for pointing that out. It seems like you will need to manually build tensorflow-text from source to run on macos. Here is a [guide](https://medium.com/@murphy.crosby/building-tensorflow-and-tensorflow-text-on-a-m1-mac-9b90d55e92df) which may help with this process. There are discussions on https://github.com/tensorflow/text as well.

**Comment by BjarneHerland** [comment]
> @BjarneHerland thank you for pointing that out. It seems like you will need to manually build tensorflow-text from source to run on macos. Here is a [guide](https://medium.com/@murphy.crosby/building-tensorflow-and-tensorflow-text-on-a-m1-mac-9b90d55e92df) which may help with this process. There are discussions on https://github.com/tensorflow/text as well.

Thanks for info @joezoug. The guide you refer to is one year old...  should be plenty of time for the official distro to catch up. An awful waste of time and cpu if everyone on arm64 must build their own wheels, IMHO  😠

**Comment by joezoug** [comment]
@BjarneHerland Just fyi, it seems like tensorflow-text has officially announced that they will not support platforms other than linux and intel-based macs: https://github.com/tensorflow/text#a-note-about-different-operating-system-packages.

I will close this issue now as it seems like the solution requires building tensorflow-text for M1 Macs.
